modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ParametersResolver.java

```diff
@@ -66,19 +71,27 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 
+import javax.inject.Inject;
+
 /**
  * Contains behavior to obtain a ResolverSet for a set of parameters values and a {@link ParameterizedModel}.
  *
  * @since 4.0
  */
 public final class ParametersResolver implements ObjectTypeParametersResolver {
 
+  @Inject
+  private ConfigurationProperties properties;
+
+  private final Boolean lazyInitEnabled;
   private final MuleContext muleContext;
   private final Map<String, ?> parameters;
 
   private ParametersResolver(MuleContext muleContext, Map<String, ?> parameters) {
     this.muleContext = muleContext;
     this.parameters = parameters;
+    inject();
```

Comment:
to inject in self on each new instance is going to hurt application start time considerably. I would aim to resolve Lazyness mode at `AbstractObjectFactory` and just pass the boolean to this constructor